src/win/util.c

```diff
@@ -1188,33 +1188,48 @@ int uv_os_homedir(char* buffer, size_t* size) {
     }
 
     bufsize = uv_utf16_to_utf8(path, -1, buffer, *size);
-    assert(len + 1 == bufsize);
     free(path);
-    *size = len;
+    *size = bufsize - 1;
 
     return 0;
   }
 
-  /* USERPROFILE is not set, so call SHGetKnownFolderPath() */
-  if (SHGetKnownFolderPath(&FOLDERID_Profile, 0, NULL, &path) != S_OK)
-    return uv_translate_sys_error(GetLastError());
+  /* USERPROFILE is not set, so call GetUserProfileDirectoryW() */
+  if (OpenProcessToken(GetCurrentProcess(), TOKEN_READ, &token) == 0) {
+    r = GetLastError();
+    free(path);
+    return uv_translate_sys_error(r);
+  }
+
+  if (!GetUserProfileDirectoryW(token, path, size)) {
+    r = GetLastError();
+    free(path);
+    CloseHandle(token);
+
+    if (r == ERROR_INSUFFICIENT_BUFFER) {
+      *size = *size - 1;
```

Comment:
This size calculation is still wrong, because we need the size in chars, not wchars. I sugesst we go the same route as in uv_cwd: https://github.com/libuv/libuv/blob/v1.x/src/win/util.c#L169 and use a stack buffer, then a call to WideCharToMultiByte without an output buffer, in order to know how big the bufffer would need to be (in utf8) and we can compare that.